In [39]:
import pandas as pd
dataset=pd.read_csv("Placement_Data_Full_Class.csv")


In [40]:
dataset

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,211,M,80.60,Others,82.00,Others,Commerce,77.60,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.00,Others,60.00,Others,Science,72.00,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.00,Others,67.00,Others,Commerce,73.00,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.00,Others,66.00,Others,Commerce,58.00,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0


In [41]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [42]:
Quan=[]
Qual=[]

In [43]:
for columnName in dataset.columns:
    if dataset[columnName].dtype=='O':
        Qual.append(columnName)
    else:
        Quan.append(columnName)

In [44]:
Quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [52]:
descrptive=pd.DataFrame(index=["Mean","Medain","Mode"],columns=Quan)

In [53]:
descrptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Medain,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Mode,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [54]:
for columnName in Quan:
    descrptive[columnName]["Mean"]=dataset[columnName].mean()
    descrptive[columnName]["Medain"]=dataset[columnName].median()
    descrptive[columnName]["Mode"]=dataset[columnName].mode()[0]


C:\Users\USER\AppData\Local\Temp\ipykernel_7380\2483475890.py:2: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descrptive[columnName]["Mean"]=dataset[columnName].mean()
C:\Users\USER\AppData\Local\Temp\ipykernel_7380\2483475890.py:3: FutureW

In [55]:
descrptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Medain,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0


Details About this Table 

The dataset contains academic percentages (SSC, HSC, Degree, MBA), 
employability test scores, and placement salaries for 108 students. 
Average academic scores are in the mid‑60s, employability test scores average ~72, 
and MBA scores average ~62. Placement salaries cluster around 2.65–3 lakh, with a mean of ~2.9 lakh, 
indicating a slightly skewed distribution. This suggests stable academic performance across levels, 
variability in test scores, and a concentration of salary offers at 3 lakh.

- ssc_p (Secondary school %)
  
- Mean ≈ 67.3, Median = 67, Mode = 62
→ Students generally scored in the mid‑60s, with a slight clustering around 62.


- hsc_p (Higher secondary %)

- Mean ≈ 66.3, Median = 65, Mode = 63
→ Similar distribution, average performance mid‑60s.


- degree_p (Undergraduate %)

- Mean ≈ 66.4, Median = 66, Mode = 65
→ Consistent with school scores, showing stable academic performance.


- etest_p (Employability test %)

- Mean ≈ 72.1, Median = 71, Mode = 60
→ Average is higher than school/degree marks, but mode at 60 suggests many scored lower, with some high scores pulling the mean up.


- mba_p (MBA %)

- Mean ≈ 62.3, Median = 62, Mode = 56.7
→ Slightly lower than earlier education, clustered around 56–62.


- salary (Placement salary)
- Mean ≈ 288,655, Median = 265,000, Mode = 300,000
→ Average salary is ~2.9 lakh, but half the students earn ≤2.65 lakh. The most common salary is 3 lakh, suggesting many offers at that level.



 Insights:

- Central tendency: Most academic scores hover around mid‑60s, employability test scores are higher, MBA scores slightly lower. Salaries cluster around 2.65–3 lakh.
- Distribution clues:
- Mean > Median for salary → right‑skewed distribution (some high salaries raise the mean).
- Mode lower than mean for test scores → suggests a spread with many low scores but some very high ones.
- Potential relationships to explore:
- Correlation between academic performance and salary.
- Whether employability test scores predict higher salaries.
- Comparison of MBA scores vs. salary outcomes.